calicoctl/calico_ctl/pool.py

```diff
@@ -122,19 +122,58 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
-    for cidr in cidrs:
+    current_pools = client.get_ip_pools(version)
+    current_pool_nets = [IPNetwork(pool.cidr) for pool in current_pools]
```

Comment:
pool.cidr is a CIDR string from IPNetwork.cidr:
`self.cidr = IPNetwork(cidr).cidr`